packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -90,18 +97,57 @@ class BuildAotCommand extends BuildSubCommand {
       }
 
       // Build AOT snapshot.
-      final int snapshotExitCode = await snapshotter.build(
-        platform: platform,
-        buildMode: buildMode,
-        mainPath: mainPath,
-        packagesPath: PackageMap.globalPackagesPath,
-        outputPath: outputPath,
-        previewDart2: previewDart2,
-        preferSharedLibrary: argResults['prefer-shared-library'],
-        extraGenSnapshotOptions: argResults[FlutterOptions.kExtraGenSnapshotOptions],
-      );
-      if (snapshotExitCode != 0) {
-        printError('Snapshotting exited with non-zero exit code: $snapshotExitCode');
+      if (platform == TargetPlatform.ios) {
```

Comment:
I wonder if we should split this whole thing and start delegating build to a build_ios and build_android (which then deals with mac.dart and gradle.dart kinda like we do with devices? (not in this pr)

Reply:
Agreed. Though if I can get `TargetArchitecture.ios_*` working smoothly, that seems like the cleanest, then we avoid the split here altogether. Unfortunately we currently use `TargetArchitecture` to mean a couple things: all through the iOS bits of the codebase, it means the platform; for the Android bits it means the target architecture. 

Will take a look at how we can consolidate this in a followup.